src/core/biblio-db.js

```diff
@@ -251,15 +178,10 @@ export const biblioDB = {
     }
     const db = await this.ready;
     const isInDB = await this.has(type, details.id);
-    return new Promise((resolve, reject) => {
-      const store = db.transaction([type], "readwrite").objectStore(type);
-      // update or add, depending of already having it in db
-      const request = isInDB ? store.put(details) : store.add(details);
-      request.onsuccess = resolve;
-      request.onerror = () => {
-        reject(new DOMException(request.error.message, request.error.name));
-      };
-    });
+    const store = db.transaction([type], "readwrite").objectStore(type);
+    // update or add, depending of already having it in db
+    const request = isInDB ? store.put(details) : store.add(details);
```

Comment:
Maybe we should do:
Maybe just: 

```suggestion
    return isInDB ? await store.put(details) : await store.add(details);
```